proof-of-spacetime.md

```diff
@@ -0,0 +1,197 @@
+## Proof-of-Spacetime
+
+This document describes Rational-PoSt, the Proof-of-Spacetime used in Filecoin.
+
+## Rational PoSt
+
+### Definitions
+
+| Name | Value |Description |
+|------|-------|------------|
+| `POST_PROVING_PERIOD` | `2880` blocks  (~24h) | The time interval in which a PoSt has to be submitted. |
+| `POST_CHALLENGE_TIME` | `240` blocks (~2h) | The time offset at which the actual work of generating the PoSt should be started. This is some delta before the end of the `Proving Period`, and as such less then a single `Proving Period`. |
+
+{{% notice todo %}}
+**TODO**: The above values are tentative and need both backing from research as well as detailed reasoning why we picked them.
+{{% /notice %}}
+
+### High Level API
+
+#### Fault Detection
+
+Fault detection happens over the course of the life time of a sector. When the sector is for some reason unavailable, the miner is responsible to submit the known `faults` using the `AddFaults` message to the chain.
+Only faults which have been reported at challenge time, will be accounted for. If any other faults have occured the miner can not submit a valid PoSt for this proving period.
```

Comment:
"Only faults which have been reported at challenge time, will be accounted for. If a miner has faults after the challenge time, then they cannot submit a valid PoSt for this proving period."

Reply:
you'd rather have them submit no post at all than let people submit faults after the deadline? I think we're on really different pages about how this all works.

Reply:
@acruikshank technically yes, but i think that they are saying that that type of slashing would be removed, which I think is a bad idea.

Having the only downside of not submitting a post for an entire proving period be 'no power for that proving period' really damages a lot of the security that rational post is built on. That basically cuts the cost of the attack by the number of proving periods you can consecutively not submit PoSts for.